borg/_hashindex.c

```diff
@@ -220,6 +230,19 @@ int shrink_size(int current){
     return hash_sizes[i];
 }
 
+int
+count_empty(HashIndex *index)
+{   /* count empty (never used) buckets. this does NOT include deleted buckets (tombstones).
+     * TODO: if we ever change HashHeader, save the count there so we do not need this function.
```

Comment:
now as you mention mmap: maybe having no empty buckets left now and then was one of the reasons why mmapping sucked in performance when jonas tried it - because the full table scan will load everything all the time (plus the memory pressure without fadvise will kick everything out again all the time).